cloudsync/tests/test_events.py

```diff
@@ -158,6 +158,14 @@ def fake_events():
         _test_events_shutdown_impl(event_counter)
 
 
+def test_events_shutdown_walk(event_counter):
+    def fake_walk(_oid):
+        return [make_event()] * 10
+
+    with patch.object(event_counter.provider, "walk_oid", fake_walk):
+        _test_events_shutdown_impl(event_counter)
```

Comment:
The test is flaky, because it erroneously passes on the code where the check occurs after `process_event`. The test should actually fail, because processing the event would be expected to have happened twice, and the assertion that it only happens once passes. Please create a test that is not threaded, since we don't really need to test the `Runnable` class will stop when stopped, we know that works, and then we can get rid of calls to `sleep` and `wait` which are inherently unreliable with respect to returning quickly after the wakeup signal is issued. If you add `time.sleep(.2)` right after the call to `wait` in `_test_events_shutdown_impl`, this will emulate a busy machine taking a little longer to wake up after the wake up signal is received, and both your test and `test_events_shutdown_provider` fail.